### PR TITLE
Rackhd rename

### PR DESCRIPTION
**Pull request by heckj**
* rename Monorail API to RackHD API throughout content
* break apart getting_started from content in development_guide
* remove a few dead files that aren't used
* rename the docs directory monorail to rackhd and fix all the relevant links

### CODE REVIEW

File: docs/rackhd/microkernel.rst [hunk 1]
```diff
@@ -149,5 +151,6 @@ changes to this script should only have to be done in one place.
 
 .. _workflows: https://github.com/rackhd/on-tasks
 
-Please send us a note if you think this is incorrect! So log as our design contraints are preserved, we are more than open to leveraging
-existing container technology.
+Please send us a note if you think this is incorrect! So long as our design
+contraints are preserved, we are more than open to leveraging existing container
```
Comment (benbp) [inline docs/rackhd/microkernel.rst]: "contraints" (my own misspelling, haha)

**Comment by benbp** [comment]
A sad day :cry: 

At least we still have our yelp link up to Monorail (http://rackhd.readthedocs.org/en/latest/software_architecture.html?highlight=yelp#major-components).

:+1: after fixing my own spelling mistake.

**Comment by heckj** [comment]
Hah - read more carefully, the architecture calls it all out as the monorail engine. :-)

**Comment by anhou** [comment]
there is still one place need to update ,  about spelling  'mMnorail'  to 'RackHD'  of "**Install the mMnorail Packages:** " in file "docs/monorail/ubuntu_install_instructions.rst"    :-)
:+1:  after this fix

**Comment by bossidy** [comment]
Are you saying that the term "monorail" needs to be removed from the entire doc? 

For example, should "monorail engine" be changed to "workflow engine"/

**Comment by bossidy** [comment]
Personally, I don't like "Getting Started" as the section title, because it is vague and could mean anything. 

Think something like "Setting Up a Test Version of RackHD" or "RackHD: Quick Setup" would be better.

**Comment by bossidy** [comment]
But I like the idea of breaking it out into its own section....

**Comment by heckj** [comment]
@bossidy updated to `RackHD: Quick Setup`. In general we renamed most of what we'd been calling monorail internally to RackHD, but I intentionally left in the monorail engine reference.

**Comment by benbp** [comment]
@christineday just a housekeeping note, when you merge a PR it's also good to click the "Delete Branch" button afterwards, that way we don't end up with hundreds of branches over time :)

**Comment by christineday** [comment]
@benbp thanks for the reminder!!! Will do.

**Pull request merged**